src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3614,6 +3614,33 @@ bool ApplicationManagerImpl::IsSOStructValid(
   return true;
 }
 
+void ApplicationManagerImpl::UnsubscribeAppFromSoftButtons(
+    const commands::MessageSharedPtr response) {
+  using namespace mobile_apis;
+
+  const Result::eType result_code = static_cast<Result::eType>(
+      (*response)[strings::msg_params][strings::result_code].asUInt());
+  const uint32_t connection_key =
+      (*response)[strings::params][strings::connection_key].asUInt();
+  const auto function_id = static_cast<FunctionID::eType>(
+      (*response)[strings::params][strings::function_id].asInt());
+
+  ApplicationSharedPtr app = application(connection_key);
+  DCHECK_OR_RETURN(app, false);
+  if (Result::REJECTED != result_code) {
+    app->UnsubscribeFromSoftButtons(function_id);
+    LOG4CXX_DEBUG(logger_,
+                  "Application has unsubscribed from softbusttons. FunctionID: "
+                      << function_id << ", app_id:" << app->app_id());
+    return true;
```

Comment:
What is the return type of this method? Bool or void?

Reply:
@APipko Return value bool. We need that parameter for Unit tests. Was reverted in 9062a2e6bb5787c23a1e305bcbe7176ae28bfc9e.